tests/ceph_volume/test_ceph_volume_redeploy.py

```diff
@@ -70,17 +70,40 @@ def run(ceph_cluster, **kw):
     size = device_spec.get("data_devices", {}).get("size", {}).replace("GB", "")
     _add_node_volumes(nodes=osd_nodes, config=config, size=size, count=4)
 
+    # Get required host from config specs
+    nodes = config.get("spec", {}).get("placement", {}).get("hosts")
+    host = [ceph_cluster.get_nodes()[int(node[-1])].hostname for node in nodes]
+
+    # Create a spec file specific to hostname
+    specs = config.get("spec", {})
+    specs["placement"]["hosts"] = host
+
+    # Refresh ceph orch devices
+    c = {"refresh": True}
+    devices = CephAdm(installer).ceph.orch.device.ls(**c)
+    if not devices:
+        raise OperationFailedError("Devices are not re-freshed")
+
     # Generate a yaml file
-    file = create_yaml_config(node=installer, config=config)
+    file = create_yaml_config(installer, specs)
 
     # Create OSDs with yaml file
     c = {"pos_args": [], "input": file}
     CephAdm(nodes=installer, mount=file).ceph.orch.apply(**c)
 
-    # Wait for cluster health to be ok
-    wait_for_cluster_health(
-        node=installer, status="HEALTH_OK", timeout=1800, interval=30
-    )
+    # Wait for OSDs ids
+    timeout, interval = 300, 6
+    for w in WaitUntil(timeout=timeout, interval=interval):
+        out = CephAdm(installer).ceph.osd.ls()
```

Comment:
Can be changed to :
`if CephAdm(installer).ceph.osd.ls():
    break`

Reply:
This we can't really do it as out is been used again in below.

Reply:
line 104 : out is being updated again so this one will be redundant